qcodes/dataset/param_spec.py

```diff
@@ -61,45 +64,79 @@ def depends_on(self):
 
     def add_inferred_from(
             self,
-            inferred_from: Sequence[Union['ParamSpec', str]]
-    ) -> None:
+            inferred_from: Sequence[Union['ParamSpec', str]]) -> None:
         """
         Args:
             inferred_from: the parameters that this parameter is inferred_from
         """
         self._inferred_from.extend(
             p.name if isinstance(p, ParamSpec) else p
-            for p in inferred_from
-        )
+            for p in inferred_from)
 
-    def add_depends_on(
-            self,
-            depends_on: Sequence[Union['ParamSpec', str]]
-    ) -> None:
+    def add_depends_on(self,
+                       depends_on: Sequence[Union['ParamSpec', str]]) -> None:
         """
         Args:
             depends_on: the parameters that this parameter depends on
         """
         self._depends_on.extend(
             p.name if isinstance(p, ParamSpec) else p
-            for p in depends_on
-        )
+            for p in depends_on)
 
     def copy(self) -> 'ParamSpec':
         """
         Make a copy of self
         """
-        return ParamSpec(
-            self.name,
-            self.type,
-            self.label,
-            self.unit,
-            self._inferred_from,
-            self._depends_on
-        )
+        return ParamSpec(self.name, self.type, self.label, self.unit,
+                         deepcopy(self._inferred_from),
+                         deepcopy(self._depends_on))
 
     def sql_repr(self):
         return f"{self.name} {self.type}"
 
     def __repr__(self):
-        return f"{self.name} ({self.type})"
+        return (f"ParamSpec('{self.name}', '{self.type}', '{self.label}', "
+                f"'{self.unit}', inferred_from={self._inferred_from}, "
+                f"depends_on={self._depends_on})")
+
+    def __eq__(self, other):
+        if not isinstance(other, ParamSpec):
+            return False
+        attrs = ['name', 'type', 'label', 'unit', '_inferred_from',
+                 '_depends_on']
+        for attr in attrs:
+            if getattr(self, attr) != getattr(other, attr):
+                return False
+        return True
+
+    def serialize(self) -> Dict[str, Any]:
+        """
+        Write the ParamSpec as a dictionary
+        """
+        output: Dict[str, Any] = {}
+        output['name'] = self.name
+        output['paramtype'] = self.type
+        output['label'] = self.label
+        output['unit'] = self.unit
+        output['inferred_from'] = self._inferred_from
+        output['depends_on'] = self._depends_on
+        output['version'] = self.version
```

Comment:
does yaml has a convention for names of fields that are not describing the contents of an object but rather the object itself? like python has double-undescore for stull like `__class__` and `__path__`. shall we use smth like this for `version`?